.github/workflows/dev-docs.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Developer Guide
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Dev docs
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 12
+      - run: npm i -g markbind-cli@3.1.1
+      - run: |
+          cd docs
+          markbind build
+      - uses: actions/github-script@v6
+        id: check-error
+        with:
+          script: |
+            const fs = require('fs');
```

Comment:
Hi @wkurniawan07, over here I am utilizing a utility action [github-script](https://github.com/actions/github-script) to make a quick check on build status and invoke the GitHub API to log an error message. Definitely possible to use shell script but I don't see any obvious benefit to switching over.
> JS code in a configuration file like this is a hidden maintenance cost.

In the context of GitHub Actions, "JS code in a configuration file" is nothing out of the ordinary - they are called ["JavaScript Action"](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action). Here are some of the aspects that I considered:
- performance
	- given that the logic is trivial and the test runs have shown that the time taken is insignificant, it's unlikely to create any performance issues down the line.
- ease of update
	- the existing code is readable. There is a bit of "verbosity" with the console logs & comments which I can remove.
	- Teammates has a front-end (that uses HTML, CSS, JS), it's unlikely that active devs here will have difficulty with JS.

Appreciate it if you can shed light on what "hidden maintenance cost" is in this context, so that I can be motivated for the change?

Reply:
Adding code snippet in a file that does not normally expect the language is often an anti-pattern because they can be easily missed by code scanning/quality checking tools or manual audits. Granted that this is a very simple code snippet and also only used for operational purpose, but it still should not be encouraged unless there are no better alternatives, which you have found and used (and kudos for that).